SocialCareCaseViewerApi.Tests/V1/UseCase/FormSubmissionUseCaseTests.cs

```diff
@@ -694,6 +663,78 @@ public void ExecuteUpdateSubmissionToApprovedThrowsUpdateSubmissionExceptionIfAp
             act.Should().Throw<UpdateSubmissionException>()
                 .WithMessage($"Worker with email {request.EditedBy} cannot approve the submission as they created the submission");
         }
+
+        [Test]
+        public void ExecuteByQueryThrowsQueryCaseSubmissionsExceptionIfNoParametersFound()
+        {
+            var request = TestHelpers.CreateQueryCaseSubmissions();
+
+            Action act = () => _formSubmissionsUseCase.ExecuteGetByQuery(request);
+
+            act.Should().Throw<QueryCaseSubmissionsException>()
+                .WithMessage("Provide at minimum one query parameter");
+        }
+
+        [Test]
+        public void ExecuteByQueryOnlyGetsFormsWithQueriedFormId()
+        {
+            const string testFormId = "foo";
+            var request = TestHelpers.CreateQueryCaseSubmissions(formId: testFormId);
+
+            var builder = Builders<CaseSubmission>.Filter;
+            var filter = builder.Empty;
+            filter &= Builders<CaseSubmission>.Filter.Eq(s => s.FormId, testFormId);
+
+            _mockMongoGateway.Setup(m =>
+                m.LoadRecordsByFilter(It.IsAny<string>(), It.IsAny<FilterDefinition<CaseSubmission>>()));
+
+            _formSubmissionsUseCase.ExecuteGetByQuery(request, filter);
+
+            _mockMongoGateway.Verify(x =>
+                x.LoadRecordsByFilter(MongoConnectionStrings.Map[Collection.ResidentCaseSubmissions], It.IsAny<FilterDefinition<CaseSubmission>>()), Times.Once);
```

Comment:
Is there a way to improve this test to confirm the function was called with something equivalent to the filter object we constructed?

Reply:
@JohnFarrellDev  Found some details about how to convert the filters to JSON format here https://jones.bz/testing-mongodb-queries/

So using the helper method: 
```
public static class FilterDefinitionExtensions
    {
        public static string RenderToJson<TDocument>(this FilterDefinition<TDocument> filter)
        {
            var serializerRegistry = BsonSerializer.SerializerRegistry;
            var documentSerializer = serializerRegistry.GetSerializer<TDocument>();
            return filter.Render(documentSerializer, serializerRegistry).ToJson();
        }
    }
```
it's possible to setup the gateway and verify the response based on the actual filter value:

 ```
...
var expectedJsonFilter = filter.RenderToJson();

  _mockMongoGateway.Setup(m =>
      m.LoadRecordsByFilter(It.IsAny<string>(), It.Is<FilterDefinition<CaseSubmission>>(filter => filter.RenderToJson().Equals(expectedJsonFilter))));

  _formSubmissionsUseCase.ExecuteGetByQuery(request, filter);

  _mockMongoGateway.Verify(x =>
      x.LoadRecordsByFilter(MongoConnectionStrings.Map[Collection.ResidentCaseSubmissions],
      It.Is<FilterDefinition<CaseSubmission>>(filter => filter.RenderToJson().Equals(expectedJsonFilter))), Times.Once);
```
That one option I suppose.

Reply:
This worked really well, thanks!